src/cript/api/api.py

```diff
@@ -514,11 +514,14 @@ def save(self, project: Project) -> None:
         """
         try:
             self._internal_save(project)
-        except Exception as exc:
-            # TODO remove all pre-handled nodes.
+        except CRIPTAPISaveError as exc:
+            if exc.pre_saved_nodes:
+                for node_uuid in exc.pre_saved_nodes:
+                    # TODO remove all pre-saved nodes by their uuid.
+                    pass
             raise exc from exc
 
-    def _internal_save(self, node, known_uuid: Optional[Set[str]] = None) -> Optional[Set[str]]:
+    def _internal_save(self, node, known_uuid: Optional[Set[str]] = None, suppress_attributes: Optional[Dict[str, Set[str]]] = None) -> Optional[Set[str]]:
```

Comment:
can we split up `_internal_save` method into smaller pieces?

I think it is fine for now if we want to keep it to try it out and get it working, but if we can split up into smaller pieces that each do one part of the project then that is all the better because otherwise if we put all the logic for create, update, getting the uuid, etc. into one method it'll make it hard to follow and read.

For me, I try to ideally keep a function/method less than 50 lines of code. Usually when it starts to go over is when I look at the function and can see places that I can split off into their own separate functions. I usually notice that I start losing track of what a function when it's over 50+ lines and lose track of a file when it is 500+ lines.

Reply:
I am doing that already.
All the heavy lifting parts are moved to `save_helpers`.
All the is left in the internal save is the general coordination of which kind of fix is needed when.